app/controllers/transfers_controller.rb

```diff
@@ -2,11 +2,10 @@
 
 class TransfersController < ApplicationController
   before_action :set_transfer, only: %i[destroy]
-  before_action :set_from_location, except: %i[destroy]
-  before_action :set_to_location, only: %i[new create]
+  before_action :set_flavors_locations, only: %i[new create]
 
   def index
-    @transfers = @from_location.transfers.load_associations.order_by_date.order_by_flavor_name
+    @transfers = Transfer.load_associations.order_by_date.order_by_flavor_name
   end
 
   def new
```

Comment:
Let's validate for the presence of to and from location if we aren't already.

Let's also validate that they are different